src/Symfony/Component/ObjectMapper/Attribute/Map.php

```diff
@@ -22,9 +22,9 @@
 readonly class Map
 {
     /**
-     * @param string|class-string|null                                                                              $source    The property or the class to map from
-     * @param string|class-string|null                                                                              $target    The property or the class to map to
-     * @param string|bool|callable(mixed, object): bool|null                                         $if        A boolean, a service id or a callable that instructs whether to map
+     * @param string|class-string|null                                                               $source    The property or the class to map from
+     * @param string|class-string|null                                                               $target    The property or the class to map to
+     * @param class-string|string|bool|callable(mixed, object): bool|null                            $if        A boolean, a service id or a callable that instructs whether to map
```

Comment:
I'd also change the getCallable method to give higher precedence to service id, before callable.

```suggestion
     * @param string|bool|callable(mixed, object): bool|null                            $if        A boolean, a service id, a callable, a target class (interpreted in this order) that instructs whether to map
```